src/main/scala/rps/Main.scala

```diff
@@ -1,51 +1,83 @@
 package rps
 import scala.io.StdIn.readLine
 import scala.util.Random
-import io.buildo.enumero.annotations.indexedEnum
+import io.buildo.enumero.annotations.enum
 import io.buildo.enumero.{CaseEnumIndex, CaseEnumSerialization}
 import Move._
+import Result._
+import de.heikoseeberger.akkahttpcirce.ErrorAccumulatingCirceSupport._
+import io.circe.generic.auto._
+import io.buildo.enumero.circe._
+import akka.actor.ActorSystem
+import akka.http.scaladsl.Http
+import akka.http.scaladsl.model._
+import akka.http.scaladsl.server.Directives._
+import akka.stream.ActorMaterializer
+import scala.io.StdIn
 
 object Main extends App {
-  Game.play
+  WebServer.run
 }
 
 object Game {
-  def play() : Unit = {
-
-  println("Choose Rock (0),PAPER (1),SCISSOR (2)")
-  val move = CaseEnumIndex[Move].caseFromIndex(readLine())
-
-  val computerMove = generateRandom()
-  val output = move.map(m => {
-  val resultString = s"${CaseEnumSerialization[Move].caseToString(m)} VS ${CaseEnumSerialization[Move].caseToString(computerMove)}-> "
-  val result =(m, computerMove) match {
-        case (x,y) if x == y => "Draw!"
-        case (ROCK,PAPER) | (PAPER,SCISSOR) | (SCISSOR,ROCK) => "You lost!"
-        case (ROCK,SCISSOR) | (PAPER,ROCK) | (SCISSOR,PAPER)  => "You win!"
-        case _ => "Unknown combination!"
-      }    
-      s"$resultString $result"
-    }).getOrElse("Not an Option!")
-    println(output)
-  }
+  def play(move : Move) : (Move,Move,Result) = {
+    val computerMove = generateRandom()
+    val result =(move, computerMove) match {
+        case (x,y) if x == y => Draw
+        case (Rock,Paper) | (Paper,Scissors) | (Scissors,Rock) => Lose
+        case (Rock,Scissors) | (Paper,Rock) | (Scissors,Paper)  => Win
+        
+    }    
+    return (move,computerMove,result)
+}
 
-   def generateRandom() = {
+  def generateRandom() = {
     Random.nextInt(3) match {
-      case 0 => ROCK
-      case 1 => PAPER
-      case 2 => SCISSOR
+      case 0 => Rock
+      case 1 => Paper
+      case 2 => Scissors
       case _ => throw new Exception("errorissimo")
     }
-}
+  } 
 }
 
-@indexedEnum trait Move {
-  type Index = String
-  object ROCK { "0" }
-  object PAPER { "1" }
-  object SCISSOR { "2" }
+@enum trait Move {
+  object Rock 
+  object Paper 
+  object Scissors 
+}
 
+@enum trait Result {
+  object Win
+  object Lose
+  object Draw
 }
 
+object WebServer {
+  def run()= {
+
+    implicit val system = ActorSystem("rps")
+    implicit val materializer = ActorMaterializer()
+    implicit val executionContext = system.dispatcher
+
+    val route =
+      pathPrefix("rps") {
+      path("play") {
+        post {
+           entity(as[Request]) { move =>
+             complete(Response tupled Game.play(move.userMove))
+        }
+        }~ options(complete())
+      }
+      }
+    val bindingFuture = Http().bindAndHandle(route, "localhost", 8080)
+    println(s"Server online at http://localhost:8080/\nPress RETURN to stop...")
```

Comment:
You could avoid repeating host and port and use some variables in both server and print, this is also a good practice to ensure you are logging the right information... Plus, host and port are very often arguments of the service